flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvBulkWriter.java

```diff
@@ -51,13 +54,19 @@ class CsvBulkWriter<T, R, C> implements BulkWriter<T> {
         checkNotNull(mapper);
         checkNotNull(schema);
 
+        // Prevent Jackson's writeValue() method calls from closing the stream.
+        mapper.getFactory().disable(JsonGenerator.Feature.AUTO_CLOSE_TARGET);
+        mapper.disable(SerializationFeature.FLUSH_AFTER_WRITE_VALUE);
+
         this.converter = checkNotNull(converter);
         this.stream = checkNotNull(stream);
         this.converterContext = converterContext;
         this.csvWriter = mapper.writer(schema);
-
-        // Prevent Jackson's writeValue() method calls from closing the stream.
-        mapper.getFactory().disable(JsonGenerator.Feature.AUTO_CLOSE_TARGET);
+        try {
+            this.generator = csvWriter.createGenerator(stream, JsonEncoding.UTF8);
```

Comment:
The Generator manages some resources with their own buffers so I think we should `close` the generator during `finish()`. The underlying `CsvEncoder` uses a char buffer that jackson [can recycle](https://github.com/FasterXML/jackson-dataformats-text/blob/3d3165e58b90618a5fbccf630f1604a383afe78c/csv/src/main/java/com/fasterxml/jackson/dataformat/csv/impl/CsvEncoder.java#L1015) using a threadlocal pool.

Disabling `AUTO_CLOSE_TARGET` will still prevent the closing of the underlying stream.